imap_processing/lo/l0/lo_science.py

```diff
@@ -439,3 +439,47 @@ def find_valid_groups(
     ]
     valid_groups = [is_sequential(seq_ctrs) for seq_ctrs in grouped_seq_ctrs]
     return valid_groups
+
+
+def organize_spin_data(dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Organize the spin data for Lo.
+
+    The spin data is spread across 28 fields. This function
+    combines each of those fields into 2D arrays for each
+    epoch and spin.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        Lo spin data from packets_to_dataset function.
+
+    Returns
+    -------
+    dataset : xr.Dataset
+        Updated dataset with the spin data organized.
+    """
+    # Get the spin data fields
+    spin_fields = [
+        "start_sec_spin",
+        "start_subsec_spin",
+        "esa_neg_dac_spin",
+        "esa_pos_dac_spin",
+        "valid_period_spin",
+        "valid_phase_spin",
+        "period_source_spin",
+    ]
+
+    for spin_field in spin_fields:
+        packet_fields = [f"{spin_field}_{i}" for i in range(1, 29)]
+        # Combine the spin data fields along a new dimension
+        combined_spin_data = xr.concat(
```

Comment:
To make sure I'm understanding what's going on: 

At this point, the dataset contains variables like:
* "start_sec_spin_1", "start_sec_spin_2",..."start_sec_spin_28"

And you're concatting all 28 into a new variable called "start_sec_spin" - right? 

Is there any info lost by not keeping track of _which_ of the 28 fields an entry started in?

Reply:
That's correct and I don't believe so. If that information is needed, you can figure it out by the array index spin_(array index + 1)